### PR TITLE
Check comment permissions for icons and edit/delete

### PR DESCRIPTION
**Pull request by JustinHoldstock**
-Generic comment and task permissions
- `can_delete` and `can_edit`

-Permissions passed to `onDelete()` of task or comment.
-Now pass `inlineDeleteMessage` prop to `Comment` to remove task specific code.
-Checking actual permissions in `deleteComment()`.
-Added `permissions` field to comment request.

TODO: Tests

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @JustinHoldstock has signed the CLA. Thanks for the pull request!

File: src/components/ContentSidebar/ActivityFeed/activity-feed/ActiveState.js [hunk 1]
```diff
@@ -90,8 +91,8 @@ const ActiveState = ({
                                 getAvatarUrl={getAvatarUrl}
                                 // permissions are not part of task API so hard code to true
                                 permissions={{
-                                    task_edit: true,
-                                    task_delete: true
+                                    can_delete: true,
```
Comment (jeremypress) [inline src/components/ContentSidebar/ActivityFeed/activity-feed/ActiveState.js]: We can do a little better here. If the user can't comment to begin with, they definitely can't edit or delete tasks.

File: src/components/ContentSidebar/ActivityFeed/comment/Comment.js [hunk 2]
```diff
@@ -156,17 +159,12 @@ class Comment extends React.Component<Props, State> {
                                     <ReadableTime timestamp={createdAtTimestamp} relativeThreshold={ONE_HOUR_MS} />
                                 </small>
                             </Tooltip>
-                            {onEdit && canEditTasksOrComments ? <InlineEdit id={id} toEdit={toEdit} /> : null}
-                            {onDelete && canDeleteTasksOrComments ? (
+                            {onEdit && canEdit ? <InlineEdit id={id} toEdit={toEdit} /> : null}
```
Comment (jeremypress) [inline src/components/ContentSidebar/ActivityFeed/comment/Comment.js]: Don't worry about actually editing comments, its not supported in the current sidebar so we shouldn't be rendering inline edit at all

File: src/components/ContentSidebar/ContentSidebar.js [hunk 3]
```diff
@@ -868,12 +868,14 @@ class ContentSidebar extends PureComponent<Props, State> {
...
      * @private
      * @param {string} commentId - The comment's id
+     * @param {Object} permissions - The comment's permissions
      * @param {Function} successCallback - the function which will be called on success
      * @param {Function} errorCallback - the function which will be called on error
      * @return {void}
      */
     deleteComment = (
         commentId: string,
+        permissions: Object,
```
Comment (jeremypress) [inline src/components/ContentSidebar/ContentSidebar.js]: Are tasks going to be in a separate PR? Also even if we aren't going to use editComment right now we should pass in the permissions object.

File: src/components/ContentSidebar/ActivityFeed/activity-feed/ActiveState.js (hunk 1 above)
Comment (JustinHoldstock) [reply src/components/ContentSidebar/ActivityFeed/activity-feed/ActiveState.js]: As we discussed, a task has a been made and we'll update in a future PR after deliberation.

File: src/components/ContentSidebar/ActivityFeed/comment/Comment.js (hunk 2 above)
Comment (JustinHoldstock) [reply src/components/ContentSidebar/ActivityFeed/comment/Comment.js]: This is required for Tasks, though. The task component contains a Comment to render text.

File: src/components/ContentSidebar/ContentSidebar.js (hunk 3 above)
Comment (JustinHoldstock) [reply src/components/ContentSidebar/ContentSidebar.js]: Tasks will be a separate PR, after deliberation with @pramodsum.

I'll update `editComment`

**Comment by boxcla** [comment]
# Webpack Bundle Size Github Bot

| Bundle        | Diff | Percentage Diff | Before | After |
| ------------- |:----:|:---------------:|:------:|:-----:|
| Total | 0 B | 0.00% | 0 B | 0 B |

Compared against previously merged PR #358

**Powered by [webpack-bundle-size-github-bot](https://github.com/wenboyu2/webpack-bundle-size-github-bot/)**
